phoenicis-engines/src/main/java/org/phoenicis/engines/EngineSettingsManager.java

```diff
@@ -35,49 +35,52 @@
  * Manages the engine settings
  */
 public class EngineSettingsManager {
-    private final PhoenicisScriptEngineFactory phoenicisScriptEngineFactory;
+    private final TypedScriptEngineFactory typedScriptEngineFactory;
     private final ExecutorService executorService;
 
     /**
      * Constructor
      *
-     * @param phoenicisScriptEngineFactory The used script engine factory
-     * @param executorService The executor service to allow for parallelization
+     * @param typedScriptEngineFactory The used script engine factory
+     * @param executorService          The executor service to allow for parallelization
      */
-    public EngineSettingsManager(PhoenicisScriptEngineFactory phoenicisScriptEngineFactory,
-            ExecutorService executorService) {
+    public EngineSettingsManager(TypedScriptEngineFactory typedScriptEngineFactory,
```

Comment:
This class shows an example how I intend to use the `script engine` in the future.

Reply:
Does it mean that we cannot access e.g. applications and engines via the same script engine?

Reply:
Yes. We will need to provide a dedicated `TypedScriptEngine` implementation for every application type (i.e. `Installer`, `Engine`, etc)

Reply:
Understood. So this is also because we want to keep the APIs in the modules and not centralize them in phoenicis-scripts (we discussed this somewhere else).

Reply:
> So this is also because we want to keep the APIs in the modules and not centralize them in phoenicis-scripts (we discussed this somewhere else).

No the reason why I want to implement a dedicated `TypedScriptEngine` class for every application type is because it divides the functionality better. It follows the approach divide and conquer in that we implement independent behavior in different modules/classes. For example what have the processing of `Installer` script and `Verb` script in common except that they are both executed with graaljs and use the module system? Therefore we divide the corresponding functionality in different classes.

I still plan to move all application types (and ideally also the `TypeScriptEngine` implementations) to the `phoenicis-scripts` module. I didn't do this yet because I think the main changes of this PR are easier comprehensible if I leave as much of the related refactoring as possible for later.

Reply:
> I still plan to move all application types

I thought @qparis doesn't like this?